src/enumerable.cr

```diff
@@ -1556,6 +1556,9 @@ module Enumerable(T)
     {% elsif T < Array %}
       # optimize for array
       flat_map &.itself
+    {% elsif T < Slice && @type < Indexable %}
+      # optimize for slice
+      Slice.join(self)
```

Comment:
Hmm, this is really surprising behavior for me, but given that it is done for arrays I guess it make sense. Where does support for Array come from? The optimization comes from c2c4d5ddbb18ad but I do wonder if it was intentional that it worked as a synonym of flatten in the first place.

Reply:
The default argument of Ruby's `Enumerable#sum` is `0`; Ruby does not have the notion of `.additive_identity` or try to infer it. This works:

```ruby
[[1, 2], [3, 4]].sum([]) # => [1, 2, 3, 4]
```

only that there is probably no optimization here.